node/src/neighborhood/mod.rs

```diff
@@ -266,38 +266,37 @@ impl Handler<ConnectionProgressMessage> for Neighborhood {
     type Result = ();
 
     fn handle(&mut self, msg: ConnectionProgressMessage, ctx: &mut Self::Context) -> Self::Result {
-        if let Ok(connection_progress) = self
+        match self
             .overall_connection_status
-            .get_connection_progress_by_ip(msg.peer_addr)
+            .get_connection_progress_to_modify(&msg)
         {
-            OverallConnectionStatus::update_connection_stage(
-                connection_progress,
-                msg.event.clone(),
-                &self.logger,
-            );
-            match msg.event {
-                ConnectionProgressEvent::TcpConnectionSuccessful => {
-                    self.send_ask_about_debut_gossip_message(ctx, msg.peer_addr);
-                }
-                ConnectionProgressEvent::IntroductionGossipReceived(_)
-                | ConnectionProgressEvent::StandardGossipReceived => {
-                    self.overall_connection_status
-                        .update_ocs_stage_and_send_message_to_ui(
-                            OverallConnectionStage::ConnectedToNeighbor,
-                            self.node_to_ui_recipient_opt
-                                .as_ref()
-                                .expect("UI Gateway is unbound"),
-                            &self.logger,
-                        );
+            Ok(connection_progress) => {
+                OverallConnectionStatus::update_connection_stage(
+                    connection_progress,
+                    msg.event.clone(),
+                    &self.logger,
+                );
+                match msg.event {
+                    ConnectionProgressEvent::TcpConnectionSuccessful => {
+                        self.send_ask_about_debut_gossip_message(ctx, msg.peer_addr);
+                    }
+                    ConnectionProgressEvent::IntroductionGossipReceived(_)
+                    | ConnectionProgressEvent::StandardGossipReceived => {
+                        self.overall_connection_status
+                            .update_ocs_stage_and_send_message_to_ui(
+                                OverallConnectionStage::ConnectedToNeighbor,
+                                self.node_to_ui_recipient_opt
+                                    .as_ref()
+                                    .expect("UI Gateway is unbound"),
+                                &self.logger,
+                            );
+                    }
+                    _ => (),
                 }
-                _ => (),
             }
-        } else {
-            trace!(
-                self.logger,
-                "An unnecessary ConnectionProgressMessage received from IP Address: {:?}",
-                msg.peer_addr
-            );
+            Err(e) => {
+                trace!(self.logger, "{}", e);
```

Comment:
Is this error specific enough to be able to track down if we see it in the log? Should an identifying string be added so that a codebase text search can lead us here?

Reply:
I also wanted to add some string to this trace log before raising the PR but somehow, I didn't. Later, when you added this comment, I got a bit excited to rephrase these logs. 

I've used this prefix to easily find the log inside our codebase - `"Found unnecessary connection progress message"`.